### PR TITLE
Move ghc-lib{,-parser} to bazel-haskell-deps

### PR DESCRIPTION
**Pull request by akrmn**
### Pull Request Checklist

- [ ] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/main/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [ ] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description
- [ ] If you mean to change the status of a component, please make sure you keep [the Component Status page](https://github.com/digital-asset/daml/blob/main/docs/source/support/component-statuses.rst) up to date.

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

**Comment by akrmn** [comment]
Ignore the fact that this currently targets the wrong branch :)

File: bazel-haskell-deps.bzl [hunk 1]
```diff
@@ -17,6 +17,12 @@ load("@os_info//:os_info.bzl", "is_linux", "is_windows")
 load("@dadew//:dadew.bzl", "dadew_tool_home")
 load("@rules_haskell//haskell:cabal.bzl", "stack_snapshot")
 
+GHC_LIB_REV = "0f0fb98205a8df24d0a4c4b869345c59"
+GHC_LIB_SHA256 = "ff32aee85a7c399814912ac5d3f3a7060860d3159f06fbb7ed8b96f326cc2c86"
+GHC_LIB_VERSION = "8.8.1.20211118"
```
Comment (akrmn) [inline bazel-haskell-deps.bzl:22]: Hm, the date at the end of this version is going to be a bit painful to work with. We could print it from `ci/da-ghc-lib/compile.yml`, but the problem is that it also ends up in `stackage_snapshot.json` and `stackage_snapshot_windows.json`, so we would have to run `bazel run @stackage-unpinned//:pin` for those anyway (they have a big `__GENERATED_FILE_DO_NOT_MODIFY_MANUALLY` at the top), defeating the purpose of the PR.

I tried setting the `version` arg in the call to `haskell_cabal_library` to a constant `"8.8.1"` (no trailing date) but it doesn't seem to change the resulting `stackage_snapshot.json`

Comment (akrmn) [reply bazel-haskell-deps.bzl]: I ended up patching `ghc-lib`'s `CI.hs` script to take a version suffix, which, if provided, replaces the date at the end. This way, the version is just `8.8.1`, so `stackage_snapshot.json` doesn't need to be updated

**Review by cocreature — APPROVED** [review]
Very nice, thanks a lot!

**Pull request merged**